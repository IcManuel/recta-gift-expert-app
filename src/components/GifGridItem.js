import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, irl }) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={irl} alt={title} />

            <p>{title}</p>
        </div>
    );
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    irl: PropTypes.string.isRequired,
}
