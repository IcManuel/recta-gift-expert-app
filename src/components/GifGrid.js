/*import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";*/
import { GifGridItem } from "./GifGridItem";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {


    const { data: images, loadding } = useFetchGifs(category);


    //getGifs();
    return (
        <>
            <h3>{category} </h3>
            {loadding && <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="card-grid animate__animated animate__fadeIn">
                {

                    images.map(img => {
                        return <GifGridItem
                            key={img.id}
                            {...img} />
                    })
                }
            </div>
        </>

    );
};