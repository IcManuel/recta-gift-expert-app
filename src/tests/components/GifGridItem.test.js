import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from "enzyme";
import React from 'react'


describe('Pruebas en el GifGridItem', () => {
    const titule = 'Un título';
    const url = 'https://localhost/ALGO.PNG';
    let wrapper = shallow(<GifGridItem irl={url} title={titule} />);



    beforeEach(() => {
        wrapper = shallow(<GifGridItem irl={url} title={titule} />);
    });

    test('Primer punto', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titule);
    });
    test('Debe de tener la imagen  igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titule);
    });

    test('Debe tener animate_fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn');
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    });


});
