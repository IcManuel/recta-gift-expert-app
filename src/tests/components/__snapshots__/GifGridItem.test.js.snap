// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGridItem Primer punto 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/ALGO.PNG"
  />
  <p>
    Un título
  </p>
</div>
`;
